### PR TITLE
`azurerm_logic_app_workflow` - add supports for `workflow_parameters`

### PR DESCRIPTION
**Pull request by magodo**
Add supports for the `workflow_parameters` (the paramter definition),
and also make the `parameters` (the paramter value) to support different
kinds of types (previously it only supports `String`).

Fixes: #7241, #11505

### CODE REVIEW

File: azurerm/internal/services/logic/logic_app_workflow_resource_test.go [hunk 1]
```diff
@@ -283,3 +298,66 @@ resource "azurerm_logic_app_workflow" "test" {
...
+      foo = "bar"
+    })
+  }
+
+  lifecycle {
+    ignore_changes = [
+      parameters["secstr"],
+      parameters["secobj"],
+    ]
+  }
```
Comment (katbyte) [inline azurerm/internal/services/logic/logic_app_workflow_resource_test.go:360]: how come we are ignoring changes heres?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: azurerm/internal/services/logic/logic_app_workflow_resource_test.go (hunk 1 above)
Comment (magodo) [reply azurerm/internal/services/logic/logic_app_workflow_resource_test.go:360]: These are secure types, means they will not return from API on read.

Comment (katbyte) [reply azurerm/internal/services/logic/logic_app_workflow_resource_test.go:360]: so in that case we should be pulling the existing values from the config through into the read function so the user does not have to do this

Comment (magodo) [reply azurerm/internal/services/logic/logic_app_workflow_resource_test.go]: You're right. I've updated the code to read the security params from state, and ignore them in the import verification accordingly. Also I refined the test for the "Array", with some modification on the code.

## Test Result

```
💤 TF_ACC=1 go test -timeout=3h -v ./azurerm/internal/services/logic -run='TestAccLogicAppWorkflow_parameters'
=== RUN   TestAccLogicAppWorkflow_parameters                                                             
=== PAUSE TestAccLogicAppWorkflow_parameters                                                             
=== CONT  TestAccLogicAppWorkflow_parameters                                                             
--- PASS: TestAccLogicAppWorkflow_parameters (92.06s)                  
PASS                                                                                                     
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/logic       92.080s
```

**Review by katbyte — APPROVED** [review]
Thanks @magodo - LGTM ❄️ - good to merge once the conflicts are resolves

**Comment by magodo** [comment]
@katbyte Thanks for the reminder, merged with the master now.

**Review by katbyte — APPROVED** [review]
Thanks @magodo - LGTM ⛰️

**Pull request merged**